vault/identity_store_oidc.go

```diff
@@ -556,6 +575,34 @@ func (i *IdentityStore) pathOIDCReadKey(ctx context.Context, req *logical.Reques
 	}, nil
 }
 
+// getRolesReferencingTargetKeyName returns a slice of roles referenced by targetKeyName.
+// Note: this is not threadsafe. It is to be called with Lock already held.
+func (i *IdentityStore) getRolesReferencingTargetKeyName(ctx context.Context, req *logical.Request, targetKeyName string) ([]role, error) {
```

Comment:
It looks like this logic was extracted from `handleOIDCDeleteKey`. Can we replace that block with a call to this function to avoid duplication?

Reply:
I actually started to replace that block with this function. But then I realized that code block creates a slice of strings of role names. Whereas here I need a slice of `roles` to be able to access each TTL.

Reply:
That's fair. If we do want to use the same func, I wouldn't mind if we were to iterate over the `[]role` returned by this func and constructing a `[]string` from that though. It should be relatively cheap since we don't have to access physical storage this time around.

Reply:
I might be missing something. But I think to do that I would need to add a `name` field to the `role struct` and then inject the name when iterating over the roleNames in the `getRolesReferencingTargetKeyName` func. I hesitate to do that. Alternatively, I could create a wrapper struct around the `role struct`. But that seems a bit unnecessary for a usage by a single function.

Reply:
Ah, you're right, the `role` doesn't hold the role name! I wonder if we use and return a map to hold both.